Warp.WebApp/Data/Redis/KeydbStorage.cs

```diff
@@ -33,23 +33,33 @@ public void Remove<T>(string key)
     }
 
 
-    public async Task Set<T>(string key, T value, TimeSpan expiresIn)
+    public async Task Set<T>(string key, T value, TimeSpan expiresIn, CancellationToken cancellationToken = default)
     {
         var db = GetDatabase<T>();
         var bytes = JsonSerializer.SerializeToUtf8Bytes(value);
         await db.StringSetAsync(key, bytes, expiresIn);
+        cancellationToken.ThrowIfCancellationRequested();
```

Comment:
I think we should use the same `WhenAny` strategy here at `Contains` and at `AddAndGetCounter` as well as at `TryGet`.